lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -10,10 +10,20 @@ class << ActiveRecord::Base
   #   act more like an array in its indexing.
   # * +add_new_at+ - specifies whether objects get added to the :top or :bottom of the list. (default: +bottom+)
   #                   `nil` will result in new items not being added to the list on create.
+  # * +sequential_updates+ - specifies whether insert_at should update objects positions during shuffling
+  #   one by one to respect position column unique not null constraint. 
+  #   Defaults to true if position column has unique index, otherwise false.
+  #   If constraint is <tt>deferrable initially deferred<tt>, overriding it with false will speed up insert_at.
   def acts_as_list(options = {})
-    configuration = { column: "position", scope: "1 = 1", top_of_list: 1, add_new_at: :bottom}
+    configuration = { column: "position", scope: "1 = 1", top_of_list: 1, add_new_at: :bottom, sequential_updates: false }
     configuration.update(options) if options.is_a?(Hash)
 
+    if options[:sequential_updates].nil?
```

Comment:
Can you move this to your method definer so that it's all in one place, and also see if this can be lazily evaluated the first time it's needed, otherwise you're forcing developers to have a valid database connection at the time the app boots which Rails doesn't do by default as far as I know. I've been bitten by this in the past. `authlogic` does this too and probably shouldn't.